config/watcher_test.go

```diff
@@ -47,17 +47,21 @@ func TestWatcher(t *testing.T) {
 
 	// Write to a different file
 	ioutil.WriteFile(filepath.Join(tempDir, "unrelated"), []byte("data"), 0644)
-	select {
-	case <-called:
-		t.Fatal("callback should not have been called for unrelated file")
-	case <-time.After(1 * time.Second):
-	}
+	require.False(t, wasCalled(called, 1*time.Second), "callback should not have been called for unrelated file")
 
 	// Write to the watched file
 	ioutil.WriteFile(f.Name(), []byte("data"), 0644)
+	require.True(t, wasCalled(called, 5*time.Second), "callback should have been called when file written")
+}
+
+// wasCalled reports whether a given callback channel was called
+// within the specified time duration or not.
+func wasCalled(c chan bool, duration time.Duration) bool {
```

Comment:
I like this helper method a lot :+1: What do you think about moving it into the `utils/testutils` package, so other test can also use it?

Reply:
Sounds good. Sending an update.

Reply:
PTAL.